### DISCUSSION TITLE
Skills de Back-end necessárias para um Dev Front-end.

### DISCUSSION THREAD

**Discussion by manzoliric** [frontendbr/forum#1049] [Discussão]
Li um topico no "Back-end BR" que era "Skills de front-end necessárias para um dev back-end" e como estou tentando arrumar um emprego com Desenvolvedor Front-end fiquei curioso o que um Front-end deve saber de Back-end?

Banco de Dados? RESTful APIs? Docker?

**Comment by felipefialho** [comment]
Linkando: https://github.com/backend-br/forum/issues/60

🎉 🎉 🎉

**Comment by wmartins** [comment]
Pra mim, a pessoa deve saber "se virar" no sentido de não depender de uma outra pessoa para executar as tarefas.

Por exemplo, subir o ambiente local e debugar minimamente caso tenha alguma falha. No sentido de, em um eventual problema, já vir com o problema mais "mastigado": "o meu banco não foi migrado, preciso de ajuda para migrar" é melhor do que "não estou conseguindo subir a aplicação".

Obviamente, se a pessoa conseguir mexer efetivamente no backend, fica melhor ainda. Exemplificando, digamos que seja necessário vir um dado extra da API para ser consumido no front. Se a pessoa conseguir mexer no backend, mesmo que adicionando um dado mockado, vai se destravar e já vai adiantar um pouco o trabalho do backend.

Sobre as tecnologias, acho que vai variar de projeto pra projeto. Saber um pouco de toda a stack do projeto ajuda para ao menos poder pesquisar como resolver os problemas.

Alguns conceitos que podem ajudar:

- Bancos de dados (SQL e NoSQL)
- Sistemas de filas/mensagens
- Caching
- Comunicação entre aplicações (REST, gRPC e etc.)
- Processo de build/deployment

**Comment by weltonsanches** [comment]
O mínimo:

- Saber enviar informações para o servidor. Receber as informações do servidor e renderizar no navegador. Pra isso é legal ter uma noção de Rest e métodos HTTP.

Tenho visto muito front usando o Wordpress pra gerenciar conteúdo, nesse caso é bom aprender o **básico** de PHP e MySQL.

Tem muita coisa legal pra saber, mas o necessário é isso. Já existe tanta coisa pra aprender no front acho que dizer que um monte de coisa de back-end é **necessário** acaba dificultando ainda mais a entrada de novos devs no mercado.

**Comment by ninetails** [comment]
Vai de empresa a empresa, depende de requisitos... Tem empresas que tem separação de trabalhos bem definidos, tem outras que querem um [fullstack](https://github.com/frontendbr/forum/issues/506)...

Para mim, acho que não é o conhecimento em backend em si, mas acho muito necessário um front hoje saber coisas simples de DevOps. Linux (UNIX) é imprescindível, pois mesmo que a pessoa não for configurar webpack ou gulp, fatalmente vai cair em configuração de ambiente, testes etc.

Em relação a conhecimento de backend em si, isso é definido pela senioridade do candidato ou da vaga. Se a vaga não pede fullstack mas pede um Junior, pelo menos de front o cara tem que saber mínimamente bem (não estou falando de saber do trabalho em si, mas dos conceitos de HTML, CSS e JS), o restante será ensinado. E também vagas que requerem o candidato saber de back, geralmente é especificado na própria vaga e não acho que há um padrão. Acho que a única coisa que diria que seria necessário seria saber sobre REST e Ajax, pq disso o candidato já teria base para a maioria das outras integrações que um sistema web poderiam ter (dizendo no caso mais genérico).

**Comment by leticiacostadias** [comment]
Acredito que isso depende muito do quão independente você quer ou precisa ser. Num time reduzido, talvez o profissional back end não consiga resolver todas as demandas que chegam pra ele, nesses casos, vale a pena estudar um pouco mais pra realmente botar a mão na massa quando for preciso.

Idealmente, num time completo e com funções muito bem definidas, talvez o front não precise saber mais do que teoria (fundamentos de REST, HTTP, DB, etc). No fim, a profundidade do teu conhecimento deveria variar cenário a cenário.

Para começar, eu me focaria nesses assuntos que são base pra você entender o que está acontecendo do outro lado.

**Comment by carloap** [comment]
O que acho de skill válida para o front-end + back, é conhecer sobre REST e SOAP, por ser possível implantar na camada de server ou de client, em alguns casos.

Saber configurar um servidor web sem ajuda, ou seja, conhecer sobre as diretivas como, configuração do "CORS", habilitar o SSL ou o cURL. Saber as diferenças e peculiaridade por exemplo, de um Tomcat pra um Apache.

Limpar e validar variáveis na camada do servidor e executar as operações CRUD de Banco de dados, sem se aprofundar em permissões.

Saber quando usar variáveis de ambiente, sessões, cookies...

Acho que isso se encaixaria para o caso de conseguir dar um help a um time ágil e fluir as tarefas. Depende de cada empresa, mas não que seja a função principal do front saber de tudo.

**Comment by rodrigoreeis** [comment]
Pelo que entendi você está procurando um emprego na área de front-end, pressupõem que você não tenha experiência na área, acredito que você está procurando vagas como Júnior/estágio, se esse for realmente seu foco no momento, não se preocupe com back end no momento, você aprenderá coisas básicas no back com experiências como essas, se for seu primeiro emprego na área de front-end meu conselho seria foque em HTML5/CSS3 e javascript.
